src/shared/Core/GenericHostProvider.cs

```diff
@@ -125,6 +126,19 @@ public override async Task<ICredential> GenerateCredentialAsync(InputArguments i
             return await _basicAuth.GetCredentialsAsync(uri.AbsoluteUri, input.UserName);
         }
 
+        public override async Task<ICredential> GetCredentialAsync(InputArguments input)
+        {
+            var credential = await base.GetCredentialAsync(input);
+            if (WebToken.IsExpiredToken(credential.Password))

```

Comment:
For readability, I think the following would be clearer:

```suggestion
            if (WebToken.TryParse(credential.Password, out var token) && token.IsExpired())
```

Reply:
Changed `TryParse` result to boolean and with new `out` parameter.
Obvious, but I just did not think of the _correct_ compact form for C#. ☹